packages/cardano-services/src/NetworkInfo/DbSyncNetworkInfoProvider/DbSyncNetworkInfoProvider.ts

```diff
@@ -43,14 +46,33 @@ export class DbSyncNetworkInfoProvider extends DbSyncProvider(RunnableModule) im
     super(db, cardanoNode, 'DbSyncNetworkInfoProvider', logger);
     this.#logger = logger;
     this.#cache = cache;
+    this.#currentEpoch = 0;
     this.#epochMonitor = epochMonitor;
     this.#builder = new NetworkInfoBuilder(db, logger);
     this.#genesisDataReady = loadGenesisData(cardanoNodeConfigPath);
   }
 
   public async ledgerTip(): Promise<Cardano.Tip> {
     const tip = await this.#builder.queryLedgerTip();
-    return toLedgerTip(tip);
+    const result = toLedgerTip(tip);
+
+    // Perform computation only on changed tip
+    if (this.#currentHash !== result.hash) {
+      this.#currentHash = result.hash;
+
+      const slotEpochCalc = createSlotEpochCalc(await this.eraSummaries());
+      const currentEpoch = slotEpochCalc(result.slot);
```

Comment:
`ledgerTip` is the most frequently used endpoint, so it would be nice to optimize this a little bit. I suggest making a change in `core` package to [memoize](https://lodash.com/docs/#memoize) both `createSlotEpochCalc` and `slotEpochCalc`

Reply:
Thank you!

Reply:
@mkazlauskas , this should be solved in 54e09fa9c